sos/report/plugins/process.py

```diff
@@ -88,4 +89,28 @@ def setup(self):
             "pidstat -p ALL -rudvwsRU --human -h",
             "pidstat -tl"
         ])
+
+    def collect(self):
+        with self.collection_file('pids_to_packages.json') as pfile:
+            if not self.policy.package_manager.query_path_command:
+                pfile.write('Package manager not configured for path queries')
+                return
+            _ps = self.exec_cmd('ps --no-headers aex')
+            pidpkg = {}
+            paths = {}
+            if not _ps['status'] == 0:
+                pfile.write(f"Unable to get process list: {_ps['output']}")
+                return
+            for proc in _ps['output'].splitlines():
+                proc = proc.strip().split()
+                pid = proc[0]
+                path = proc[4]
+                if not self.path_exists(path):
+                    continue
+                if path not in paths:
+                    paths[path] = self.policy.package_manager.pkg_by_path(path)
```

Comment:
Calling external packaging command (like, say, `rpm -qf <binary>`) for each and every binary *separately* takes some time. I suggest (once reviewed and improved) idea "call rpm -qf on all binaries once, parse the output", as that is quite faster.

I did some comparison on a small RHEL8 system and also on a bit busy one (320 processes), comparing `sos report -o process --batch --build` execution times for:
1) `process` plugin empty (`return` at the beginning of `setup()` and `collect()` methods)
2) the plugin calling just `collect()` from this PR
3) the plugin calling just `collect()` which calls the package query once

idle RHEL8 system, average values from 30 tests:
- no `collect`: `3.15687s`
- `collect` from this PR: `3.8576s`
- `collect` from my suggestion: `3.27247s`

busier RHEL system, average values from 30 tests:
- no `collect`: `3.65807s`
- `collect` from this PR: `4.60553s`
- `collect` from my suggestion: `3.8297s`


My `collect()` method (which relies three lists have always the same length, maybe too fragile construction?). The implementation does contain identical paths multiple times within the one query (if multiple processes run the same binary), yet still it is faster than original approach:

    def collect(self):
        with self.collection_file('pids_to_packages.json') as pfile:
            if not self.policy.package_manager.query_path_command:
                pfile.write('Package manager not configured for path queries')
                return
            _ps = self.exec_cmd('ps --no-headers aex')
            pidpkg = {}
            paths = {}
            if not _ps['status'] == 0:
                pfile.write(f"Unable to get process list: {_ps['output']}")
                return
            paths = []
            pids = []
            for proc in _ps['output'].splitlines():
                proc = proc.strip().split()
                pid = proc[0]
                path = proc[4]
                if not self.path_exists(path):
                    continue
                pids.append(pid)
                paths.append(path)

            pkgs = self.policy.package_manager.pkg_by_path(' '.join(paths))
            for i in range(len(pids)):
                pidpkg[pids[i]] = {'path': paths[i], 'package': pkgs[i]}
            pfile.write(json.dumps(pidpkg, indent=4))

Reply:
Testing this out, multiple packages for the same path seem to throw this off due to the requirement of equal-length lists:

```
    "1208": {
        "path": "/usr/bin/python3",
        "package": "python3-3.10.6-1.fc35.x86_64"
    },
... later ...
    "3657": {
        "path": "/usr/bin/python3",
        "package": "evolution-data-server-3.42.5-2.fc35.x86_64"
    },


and from CLI:

$ rpm -qf /usr/bin/python3
python3-3.10.6-1.fc35.x86_64
python3-3.10.6-1.fc35.i686
```

Reply:
I see - the argument "multiple packages output" from elsewhere breaks my idea, then. Sad to see the improvement wont work (and ACK to the whole PR itself, then).